langchain/document_loaders/onedrive.py

```diff
@@ -0,0 +1,219 @@
+"""Loader that loads data from OneDrive"""
+
+import logging
+import os
+import tempfile
+from enum import Enum
+from pathlib import Path
+from typing import Dict, List, Optional, Type, Union
+
+try:
+    from O365 import Account, FileSystemTokenBackend
```

Comment:
because this is optional, we don't want to do this at top of file (we only want to do it when needed). 

if this is needed for linting, you can do

```
# top of file
from __future__ import annotations

from typing import TYPE_CHECKING
...

if TYPE_CHECKING:
     from O365 import...
```

if this is not for type hinting only, but rather you want to use it, you should do the imports in the functions themselves where they are used

Reply:
Some of these imports are used within the functions and in the **function signature**. I've tried to move them into a __init__ method, but the interpreter still complains it cannot import. I am not able to make this work by moving those imports from the top. Do you have any suggestions, here?